### PR TITLE
Fix 2267

### PR DESCRIPTION
**Pull request by AlanCezarAraujo**
Associated Issue: #2267

### Summary of Changes

* On **selectResultItem** is now possible to pass the event as a parameter to close the result list

### Test Plan

- [x] Open a large file
- [x] Ctrl+f to find
- [x] Select **function** from filter
- [x] Enter the text **add**
- [x] Click on any item
- [x] The result list show close after the click

### CODE REVIEW

File: src/components/Editor/SearchBar.js [hunk 1]
```diff
@@ -399,12 +402,15 @@ const SearchBar = React.createClass({
   },
 
   // Handlers
-  selectResultItem(item: FormattedSymbolDeclaration) {
+  selectResultItem(e: SyntheticEvent, item: SymbolDeclaration, index?: number) {
```
Comment (jasonLaster) [inline src/components/Editor/SearchBar.js]: Hmm, my only question is around `FormattedSymbolDeclaration` => `SymbolDeclaration`. Why was this needed?

Comment (AlanCezarAraujo) [inline src/components/Editor/SearchBar.js]: Once I only needed the item's location, `FormattedSymbolDeclaration` would expect more data than  I needed, and it could also break the _line length_ lint rule.

Comment (irfanhudda) [inline src/components/Editor/SearchBar.js]: For _line length_ issue we can always wrap arguments to different lines like https://github.com/devtools-html/debugger.html/blob/master/src/components/Editor/ConditionalPanel.js#L10

I wonder why we have `index` function parameter. It doesn't appear anywhere in function.

Comment (AlanCezarAraujo) [inline src/components/Editor/SearchBar.js]: @irfanhudda thanks for the tip about line length!
The `index` was dead code. I just removed it.

**Comment by jasonLaster** [comment]
Thanks @AlanCezarAraujo!
